lib/iris/cube.py

```diff
@@ -930,19 +930,19 @@ def _normalise_attrs(
         return attributes
 
     @property
-    def locals(self):
+    def locals(self) -> LimitedAttributeDict:
         return self._locals
 
     @locals.setter
-    def locals(self, attributes):
+    def locals(self, attributes: Optional[Mapping]):
```

Comment:
I'm surprised to see `Optional` here. It's the setter, and this is the value you are setting. Is there some scope for setting `None` that I'm not seeing?

Reply:
Yes you can `cube.attributes.local = None`, \~equivalent to `= {}`. 
Mainly because you can already `cube.attributes = None` -- it's supported.
Technically, It's subtly different from `.attributes.clear()`